contrib/systemd/etcd2-backup-coreos/README.md

```diff
@@ -0,0 +1,253 @@
+
+You should repeat the process for `e2` and `e3`. If you do not want a node to generate backups, omit enabling and starting `etcd2-backup.timer`.
+
+## Restore the cluster
+
+Let's assume that the a mischevious friend decided it would be a good idea to corrupt the etcd2 data-dir on ALL of your nodes (e1,e2,e3). You simply want to restore the cluster from `e1`'s backup.
+
+Here's how you would recover:
+
+```sh
+# First, make SURE etcd2 and etcd2-backup are not running on any nodes
+for node in e{1..3};do
+    ssh core@$node "sudo systemctl stop etcd2.service etcd2-backup.{timer,service}"
+done
+
+ssh core@e1 "sudo systemctl start etcd2-restore.service && sudo systemctl start etcd2.service"
+
+for node in e{2..3};do
+    ssh core@$node "sudo systemctl start etcd2-join.service && sudo systemctl start etcd2.service"
+    sleep 10
+done
+```
+After e2 and e3 finish catching up, your cluster should be back to normal.
+
+## Migrate the cluster
+
+The same friend who corrupted your etcd2 data-dirs decided that you have not had enough fun. This time, your friend dumps coffee on the machines hosting e1, e2 and e3. There is a horrible smell, and the machines are dead.
+
+Luckily, you have a new 3-node etcd2 cluster ready to go, along with the s3 backup for e1 from your old cluster.
+
+The new cluster configuration looks like this. Assume that etcd2-backup is not installed. (If it is, you NEED to make sure it's not running on any nodes)
+
+| ETCD_NAME  | ETCD_ADVERTISED_PEER_URL |
+| ------------- |:-------------:|
+| q1   | http://172.17.8.201:2379 |
+| q2   | http://172.17.8.202:2379 |
+| q3   | http://172.17.8.203:2379 |
+
+We will assume `q1` is the chosen founding member, though you can pick any node you like.
+
+## Migrate the remote backup
+
+First, you need to copy your backup from `e1`'s backup folder to `q1`'s backup folder. I will show the s3 example.
+
+```sh
+# Make sure to remove q1's backup directory, if it exists already
+aws s3 rm --recursive s3://etcd2-backup-bucket/backups/<q1-machine-id>
+aws s3 cp --recursive s3://etcd2-backup-bucket/backups/<e1-machine-id> s3://etcd2-backup-bucket/backups/<q1-machine-id>
+```
+
+## Configure the new cluster.
+
+```
+[Service]
+Environment="ETCD_RESTORE_MASTER_ADV_PEER_URLS=http://172.17.8.201:2379"
+Environment="RCLONE_ENDPOINT=s3-testing-conf:s3://etcd2-backup-bucket/backups"
+```
+Remember that, since this is a new cluster, each new node will have new `machine-id` and will not clobber your backups from the old cluster, even though `RCLONE_ENDPOINT` is the same for both the old `e` cluster and the new `q` cluster.
+
+## Installation
+
+We first want to install the configured etcd2-backup package on all nodes, but not start any services yet.
+
+```sh
+cd etcd2-backup
+./build
+for node in q{1..3};do
+    scp etcd2-backup.tgz core@$node:~/
+    ssh core@$node "mkdir -p ~/etcd2-backup
+    ssh core@$node "mv etcd2-backup.tgz etcd2-backup/
+    ssh core@$node " cd etcd2-backup
```

Comment:
There is some weirdness going on with quotes and spacing in this block. Perhaps just combine into a single ssh command?